lib/fastlane/plugin/revenuecat/actions/bump_version_update_changelog_create_pr_action.rb

```diff
@@ -0,0 +1,109 @@
+
+      def self.description
+        "Bumps sdk version, asks to update changelog and creates PR with changes."
+      end
+
+      def self.authors
+        ["Toni Rico"]
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :current_version,
+                                       description: "Current version of the sdk",
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :changelog_latest_path,
+                                       description: "Path to CHANGELOG.latest.md file",
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :changelog_path,
+                                       description: "Path to CHANGELOG.md file",
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :files_to_update,
+                                       env_name: "FILES_TO_UPDATE_VERSION",
+                                       description: "Files that contain the version number and need to have it updated",
+                                       optional: false,
+                                       type: Array),
+          FastlaneCore::ConfigItem.new(key: :files_to_update_without_prerelease_modifiers,
+                                       env_name: "FILES_TO_UPDATE_VERSION_WITHOUT_PRERELEASE_MODIFIERS",
+                                       description: "Files that contain the version number without release modifiers and need to have it updated",
+                                       optional: true,
+                                       default_value: [],
+                                       type: Array),
+          FastlaneCore::ConfigItem.new(key: :repo_name,
+                                       env_name: "REPO_NAME",
+                                       description: "Name of the repo of the SDK",
+                                       optional: false,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :github_token,
+                                       env_name: "GITHUB_TOKEN",
+                                       description: "Github token to use to prepopulate the changelog",
+                                       optional: true,
+                                       type: String),
+          FastlaneCore::ConfigItem.new(key: :github_rate_limit,
+                                       env_name: "GITHUB_RATE_LIMIT_SLEEP",
+                                       description: "Sets a rate limiter for github requests when creating the changelog",
+                                       optional: true,
+                                       default_value: 0,
+                                       type: Integer),
```

Comment:
I need to redo how the changelog generator works so that the isn't needed 🙈

Reply:
When I was adding tests for that part of the process, I noticed why we have to do a request for each commit. To get the PR number. I think all the other information would be available within the response of the list of commits. I'm not sure if there is an easier way to get the PR number from each commit, but it would be great to avoid this problem :P